pkg/operator/etcdmemberscontroller/etcdmemberscontroller.go

```diff
@@ -0,0 +1,208 @@
+package etcdmemberscontroller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-etcd-operator/pkg/etcdcli"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+const workQueueKey = "key"
+
+// EtcdMembersController reports the status conditions
+// of etcd members.
+type EtcdMembersController struct {
+	operatorClient v1helpers.OperatorClient
+	etcdClient     etcdcli.EtcdClient
+
+	eventRecorder events.Recorder
+	queue         workqueue.RateLimitingInterface
+}
+
+func NewEtcdMembersController(operatorClient v1helpers.OperatorClient,
+	etcdClient etcdcli.EtcdClient,
+	eventRecorder events.Recorder,
+) *EtcdMembersController {
+	c := &EtcdMembersController{
+		operatorClient: operatorClient,
+		etcdClient:     etcdClient,
+
+		eventRecorder: eventRecorder.WithComponentSuffix("member-observer-controller"),
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "EtcdMembersController"),
+	}
+	return c
+}
+
+func (c *EtcdMembersController) sync() error {
+	err := c.reportEtcdMembers()
+	if err != nil {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "EtcdMembersDegraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "ErrorUpdatingReportEtcdMembers",
+			Message: err.Error(),
+		}))
+		if updateErr != nil {
+			c.eventRecorder.Warning("ReportEtcdMembersErrorUpdatingStatus", updateErr.Error())
+		}
+		return err
+	}
+
+	_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+		Type:   "EtcdMembersDegraded",
+		Status: operatorv1.ConditionFalse,
+		Reason: "MembersReported",
+	}))
+	if updateErr != nil {
+		c.eventRecorder.Warning("ReportEtcdMembersErrorUpdatingStatus", updateErr.Error())
+		return updateErr
+	}
+	return nil
+}
+
+func (c *EtcdMembersController) reportEtcdMembers() error {
+	etcdMembers, err := c.etcdClient.MemberList()
+	if err != nil {
+		return err
+	}
+
+	healthyEtcdMembers, notStartedEtcdMembers, unhealthyMembers, unknownEtcdMembers := []string{}, []string{}, []string{}, []string{}
+
+	for _, m := range etcdMembers {
+		switch c.etcdClient.MemberStatus(m) {
+		case etcdcli.EtcdMemberStatusAvailable:
+			healthyEtcdMembers = append(healthyEtcdMembers, m.Name)
+		case etcdcli.EtcdMemberStatusNotStarted:
+			notStartedEtcdMembers = append(notStartedEtcdMembers, m.Name)
+		case etcdcli.EtcdMemberStatusUnhealthy:
+			unhealthyMembers = append(unhealthyMembers, m.Name)
+		case etcdcli.EtcdMemberStatusUnknown:
+			unknownEtcdMembers = append(unknownEtcdMembers, m.Name)
+		}
+	}
+
+	if len(unhealthyMembers) != 0 {
```

Comment:
`|| len(unknownMembers) > 0`

Reply:
added and to be consistent changed to `len(unhealthyMembers)  > 0`